src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java

```diff
@@ -1082,4 +1087,175 @@ public void terminateEvent(final Integer procId, final String eventId, final Ter
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        String.format(ERR_MSG_FMT_SUPPLIER_NOT_FOUND, supplierId)));
+
+    Optional<Supplier> requestSupplier =
+        exportRfxResponse.getSuppliersList().getSupplier().stream()
+            .filter(
+                supplier ->
+                    supplier
+                        .getCompanyData()
+                        .getId()
+                        .equals(supplierOrganisationMapping.getExternalOrganisationId()))
+            .findFirst();
+
+    List<SupplierAttachmentResponse> supplierAttachmentResponsesList =
+        getSupplierAttachmentResponses(
+            procId, eventId, supplierOrganisationMapping.getExternalOrganisationId().toString());
+
+    return supplierAttachmentResponsesList.stream().findFirst().get();
+  }
+
+  private List<SupplierAttachmentResponse> getSupplierAttachmentResponses(
+      final Integer procId, final String eventId, final String supplierId) {
+
+    var procurementEvent = validationService.validateProjectAndEventIds(procId, eventId);
+    var exportRfxResponse = jaggaerService.getRfx(procurementEvent.getExternalEventId());
+    var supplierList = exportRfxResponse.getSuppliersList();
+
+    if (ObjectUtils.allNotNull(
+        exportRfxResponse.getOffersList(), exportRfxResponse.getOffersList())) {
+
+      List<Offer> offersWithParameters =
+          getOffersWithSupplierAttachments(exportRfxResponse, supplierId);
+      List<SupplierAttachmentResponse> supplierAttachmentResponsesList = new ArrayList<>();
+
+      for (Offer offer : offersWithParameters) {
+        var supplierAttachmentResponse =
+            SupplierAttachmentResponse.builder()
+                .supplierId(offer.getSupplierId())
+                .supplierName(getSupplierName(supplierList, offer.getSupplierId()))
+                .parameterInfoList(new ArrayList<>())
+                .build();
+        var attachmentParametersOnly =
+            offer.getTechOffer().getParameterResponses().getParameter().stream()
+                .filter(parameter -> parameter.getParameterType().equals("ATTACH"))
+                .collect(Collectors.toList());
+
+        var parameterInfoList =
+            attachmentParametersOnly.stream()
+                .map(
+                    parameter -> {
+                      return ParameterInfo.builder()
+                          .parameterId(parameter.getParameterId())
+                          .attachmentInfoList(getAttachmentInfoForParameter(parameter))
+                          .build();
+                    })
+                .collect(Collectors.toList());
+
+        supplierAttachmentResponse.getParameterInfoList().addAll(parameterInfoList);
+
+        supplierAttachmentResponsesList.add(supplierAttachmentResponse);
+      }
+      return supplierAttachmentResponsesList;
+    } else {
+      throw new IllegalArgumentException(
```

Comment:
This will result in a `400` I think - is that right or do we want to throw a `ResourceNotFoundException` resulting in a 404 instead?

Reply:
Agree. Updated